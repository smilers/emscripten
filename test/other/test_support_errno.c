/*
 * Copyright 2016 The Emscripten Authors.  All rights reserved.
 * Emscripten is available under two separate licenses, the MIT license and the
 * University of Illinois/NCSA Open Source License.  Both these licenses can be
 * found in the LICENSE file.
 */

#include <stdio.h>
#include <errno.h>
#include <string.h>
#include <emscripten/em_asm.h>
#include <emscripten/em_js.h>

EM_JS_DEPS(test, "$setErrNo");

int main() {
  EM_ASM(setErrNo(5));
  printf("errno   : %d\n", errno);
  printf("strerror: %s\n", strerror(errno));
  return 0;
}
