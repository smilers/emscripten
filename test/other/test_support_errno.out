errno   : 5
strerror: Address family not supported by protocol
